internal/dbutil/sqliteutil/migrator.go

```diff
@@ -159,20 +164,28 @@ func KeepTables(ctx context.Context, db *sql.DB, tablesToKeep ...string) error {
 	for _, tableName := range tables {
 		if !tableToKeep(tableName, tablesToKeep) {
 			// Drop tables we aren't told to keep in the destination database.
-			_, err = db.Exec(fmt.Sprintf("DROP TABLE %s;", tableName))
+			_, err = txn.Exec(fmt.Sprintf("DROP TABLE %s;", tableName))
 			if err != nil {
 				return errs.Wrap(err)
 			}
 		}
 	}
 
+	if err == nil {
+		err = errs.Wrap(txn.Commit())
+	} else {
+		err = errs.Combine(err, txn.Rollback())
+	}
+
 	// VACUUM the database to reclaim the space used by the dropped tables. The
 	// data will not actually be reclaimed until the db has been closed.
+	// We don't include this in the above transaction because
+	// you can't VACUUM within a transaction with SQLite3.
```

Comment:
This may be covered in a later commit, but do we want to vacuum even if the tx fails? It seems like we want to return the error on line 177 and abort.